### PR TITLE
Update subscription query with Event types only and improve Event picking

### PR DESCRIPTION
**Pull request by 2can**
I want to merge this change because...

* use introspection query to get WebhookEvents that implements Event #3200
* handle additions and deletions of Events with AST #3193

**PR intended to be tested with API branch:** 

### Screenshots

Mar-14-2023 13-13-05

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [x] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Acceptance Criteria
1. [ ] subscription query fragment is removed while Event is unchecked
2. [ ] subscription query is empty when all Events are unchecked
3. [ ] checking an `WebhookEvent` which doesn't implement `Event` (eg `Observability`, `NotifyUser`) doesn't update subscription query with subscription fragment
4. [ ] checking / unchecking `Events` doesn't create glitches with multiple fragments being inserted

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
5. [ ] giftCard
6. [ ] category
7. [ ] collection
8. [ ] attribute
9. [ ] productType
10. [ ] shipping
11. [ ] customer
12. [ ] permissions
13. [ ] menuNavigation
14. [ ] pages
15. [ ] sales
16. [ ] vouchers
17. [ ] homePage
18. [ ] login
19. [ ] orders
20. [ ] products
21. [x] app
22. [ ] plugins
23. [ ] translations
24. [ ] navigation
25. [ ] variants
26. [ ] payments

CONTAINERS=1

### CODE REVIEW

**Review by karola312 — APPROVED** [review]

File: src/custom-apps/components/WebhookDetailsPage/utils.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        interfaces {
+          name
+        }
+        description
+      }
+    }
+  }
+`;
+
+const isEvent = ({ name }) => name === "Event";
```
Comment (poulch) [inline src/custom-apps/components/WebhookDetailsPage/utils.ts:26]: suggestion: missing types for function params

File: src/custom-apps/handlers.ts [hunk 2]
```diff
@@ -86,62 +105,125 @@ export const createAsyncEventsSelectHandler =
...
+
+const handleQuery = ({
+  events,
+  query,
+  setQuery,
+  availableEvents,
+}: HandleQuery) => {
+  const availableEventNames = availableEvents.map(({ name }) => name);
+  const eventsNames: string[] = events
+    .map(event => enumToEventName(event))
```
Comment (poulch) [inline src/custom-apps/handlers.ts:133]:
```suggestion
    .map(enumToEventName)
```

**Review by poulch — DISMISSED** [review]

File: src/custom-apps/components/WebhookDetailsPage/WebhookDetailsPage.tsx [hunk 3]
```diff
@@ -92,6 +94,13 @@ const WebhookDetailsPage: React.FC<WebhookDetailsPageProps> = ({
     onSubmit({ ...data, ...{ subscriptionQuery: query } });
   };
 
+  const { data: introspectionData } = useQuery(IntrospectionQuery, {
+    fetchPolicy: "network-only",
+  });
```
Comment (Droniu) [inline src/custom-apps/components/WebhookDetailsPage/WebhookDetailsPage.tsx:99]: **issue (blocking):** GraphQL queries should not be in `Page` files, it breaks the convention. Please move it to the parent view component and pass data in props.

**Review by Droniu — CHANGES_REQUESTED** [review]

File: src/custom-apps/components/WebhookDetailsPage/WebhookDetailsPage.tsx (hunk 3 above)
Comment (2can) [reply src/custom-apps/components/WebhookDetailsPage/WebhookDetailsPage.tsx]: Fixed

File: src/custom-apps/handlers.ts (hunk 2 above)
Comment (2can) [reply src/custom-apps/handlers.ts]: Done

File: src/custom-apps/components/WebhookDetailsPage/utils.ts (hunk 1 above)
Comment (2can) [reply src/custom-apps/components/WebhookDetailsPage/utils.ts]: Done, utils  moved to `src/custom-apps/utils.tsx`

**Review by Droniu — DISMISSED** [review]

**Pull request merged**